packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -765,10 +778,13 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
 ///    element for the element hierarchy.
 ///  * [WidgetsBinding.handleBeginFrame], which pumps the widget pipeline to
 ///    ensure the widget, element, and render trees are all built.
-void runApp(Widget app) {
+///  * [DefaultShaderWarmUp], which will be used to warm up Skia shader
+///    compilations by default.
+void runApp(Widget app, {ShaderWarmUp shaderWarmUp = const DefaultShaderWarmUp()}) {
   WidgetsFlutterBinding.ensureInitialized()
     ..attachRootWidget(app)
     ..scheduleWarmUpFrame();
+  shaderWarmUp.execute();
 }
```

Comment:
I think now that we've moved this to the painting library, it's more clear that `runApp` is the wrong layer to call this from. I'd call this from the painting library's binding's initialization logic. Maybe have a getter there that returns the ShaderWarmUp instance to use.

You can still take a `shaderWarmUp` method here, just pass it to `WidgetsFlutterBinding.ensureInitialized` and have the WidgetsBinding implementation override the painting library's getter to return the provided value if it's not null.